src/applications/soundDetectionDemo/cochleaEventsMapper/src/vCochleaEventsMapper.cpp

```diff
@@ -0,0 +1,406 @@
+            for(int j = 0; j < CochleaEvent::mso_num_neurons_per_channel; j++) {
+                mso_histogram[i][j] = 0;
+            }
+        }
+
+        // ----------------------------------------------------------------------------
+        //                          HISTOGRAM DRAW SECTION
+        // ----------------------------------------------------------------------------
+
+        // ----------------------------------------------------------------------------
+        //                          CLEAR IMAGE
+        // Remove all the previous plot
+        image_nas_histogram = cv::Mat::zeros(cv::Size(image_nas_histogram_width, image_nas_histogram_height), CV_8UC3);
+
+        // ----------------------------------------------------------------------------
+        //                          IMAGE SETTING
+        // Define the image settings
+        float bar_width = ((float)(image_nas_histogram_width)) / CochleaEvent::nas_addrresses_offset;
+        float bar_height = 0.0f;
+
+        // Horizontal and vertical margins: set to 3% of the image height
+        float margin = image_nas_histogram.cols * 0.03;
+
+        // Calculate max and min val (for normalization) 
+        float max_nas = 0.0f;
+        float min_nas = 100000.0f;
+
+        for(int i = 0; i < CochleaEvent::nas_addrresses_offset; i++) {
+            float val = (float)(nas_histogram[i]);
+            if(val > max_nas){
+                max_nas = val;
+            }
+            if(val < min_nas){
+                min_nas = val;
+            }
+        }
+
+        // ----------------------------------------------------------------------------
+        //                          HISTOGRAM DRAW
+        for(int i = 0; i < CochleaEvent::nas_addrresses_offset; i++) {
+            // First, get the normalized value of the neuron activity
+            float normalized_histogram_value = (nas_histogram[i] - min_nas) * 100.0 / (max_nas - min_nas);
+            
+            // Define left top point
+            float rec_x_top_left = i * bar_width;
+            // The y value is the bottom reference value minus the normalized value (the width of the histogram bar)
+            float rec_y_top_left = 5 * (image_nas_histogram.rows/6) - normalized_histogram_value;
+            cv::Point p1(rec_x_top_left, rec_y_top_left);
+
+            // Define right bottom point
+            // X value is the same as left top point plus the diameter of the circle (radius * 2)
+            float rec_x_bottom_right = rec_x_top_left + bar_width;
+            float rec_y_bottom_right = 5 * (image_nas_histogram.rows/6);
+            cv::Point p2(rec_x_bottom_right, rec_y_bottom_right);
+
+            // Create the rectangle
+            cv::Vec3b c = cv::Vec3b(255, 255, 255);
+            cv::rectangle(image_nas_histogram, p1, p2, c);
+        }
+        // Show the image
+        cv::imshow("nas_histogram", image_nas_histogram);
+        cv::waitKey(10);
```

Comment:
FYI it is probably best to have all the `cv::imshow()` as you have and a single `cv::waitKey()` at the end of the function. the single `cv::waitkey()` will allow the drawing all at the same time without unecessary delays.